packages/suite/src/actions/suite/suiteActions.ts

```diff
@@ -417,6 +421,7 @@ export const forgetDevice = (payload: TrezorDevice): SuiteAction => ({
 export const createDeviceInstance =
     (device: TrezorDevice, useEmptyPassphrase = false) =>
     async (dispatch: Dispatch, getState: GetState) => {
+        const devices = selectDevices(getState());
```

Comment:
This should go just before the dispatch of SUITE.CREATE_DEVICE_INSTANCE (so after the potential await for applySettings), I guess. Devices can theoretically change while waiting for connect response.

Reply:
https://github.com/trezor/trezor-suite/pull/9202/commits/a2ef83f59dfc5af75970129c02200383412955cb